jupyter_server_ydoc/handlers.py

```diff
@@ -309,6 +321,7 @@ def on_document_change(self, event):
         self.saving_document = asyncio.create_task(self.maybe_save_document())
 
     async def maybe_save_document(self):
+        assert isinstance(self.room, DocumentRoom)
```

Comment:
This doesn't need to be done in this PR, but doing validation imperatively in these methods hinders readability, as we are adding code unrelated to the core purpose of each method. Can this not be done in the constructor? Or if `self.room` is mutable, can we abstract this away in a decorator like `validate_room`?

Reply:
It's actually the type checker that was complaining, and for a good reason. The `assert` guides the type checker, but we could also use `cast`. I think it also guides us, since we know that at this point it cannot be a `TransientRoom`.